server/src/main/java/org/opensearch/cluster/routing/allocation/decider/NodeLoadAwareAllocationDecider.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing.allocation.decider;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.routing.RoutingNode;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.allocation.RoutingAllocation;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.settings.Setting.Property;
+
+import java.util.function.BiPredicate;
+
+/**
+ * This {@link NodeLoadAwareAllocationDecider} controls shard over-allocation
+ * due to node failures or otherwise on the surviving nodes. The allocation limits
+ * are decided by the user provisioned capacity, to determine if there were lost nodes
+ * <pre>
+ * cluster.routing.allocation.overload_awareness.provisioned_capacity: N
+ * </pre>
+ * <p>
+ * and prevent allocation on the surviving nodes of the under capacity cluster
+ * based on overload factor defined as a percentage by
+ * <pre>
+ * cluster.routing.allocation.load_awareness.skew_factor: X
```

Comment:
50% skew means 25% node failures it can handle. Just thinking what is more easier for user to calculate and set? Is it how much more shards he should allow per node or how many node failures he can tolerate?

Reply:
shards per node is the common balancing factor and determines the actual performance parameters, node loss can mean data loss without any overload too. So I would say shards per node is more deterministic

Reply:
Also there is already a settings that clients are exposed to today, so maybe it is actually getting used too 
`cluster.routing.allocation.total_shards_per_node` . This setting is just another relaxed variation of this that also factors in the growing needs of the cluster also very soft on primaries.

Reply:
ok, makes sense